libraries/chain/contracts/eosio_contract.cpp

```diff
@@ -181,14 +181,15 @@ void apply_eosio_updateauth(apply_context& context) {
       if (act_auth.permission == config::owner_name || act_auth.permission == update.permission) {
          return true;
       }
+      const permission_object *current = db.find<permission_object, by_owner>(boost::make_tuple(update.account, update.permission));
+      // Permission doesn't exist yet
+      if (current == nullptr) return true;
```

Comment:
I think this has the unintended side effect of allowing _any permission_ the ability to create a permission.  Even if that permission is linked into the tree _above_ it.  For instance, if we have:
* active
  * foo

we can authorize @foo to create bar whose _parent_ is active:

* active
  * foo
  * bar

This shouldn't be possible, but we are earlying out with `true` when the permission doesn't exist.

I think instead, we should handle the case where the permission doesn't exist by setting `current` to the parent-to-be and then processing the loop completely.  

This would create a situation where a permission level can only create permissions that attach to the tree of permissions below it.

Reply:
Valid point.

PR Updated. Now, when creating a new permission, it will check whether the parent permission exists and has permission equal or lower to the carried auth.